base/settings/R/pft_site_linker.R

```diff
@@ -56,10 +56,11 @@ site.pft.linkage <- function(settings, site.pft.links) {
           "Since your site xml tag does NOT have a site id,",
           "we can not assign a PFT to it. The site of this site is",
           (site.setting[["run"]])$site$name)
-      }
-      # see if we can find this site id in the LUT
-      if (site.id %in% site.pft.links$site) {
-       site.pft <- site.pft.links$pft[which(site.pft.links$site %in% site.id)]
+      } else {
+        # see if we can find this site id in the LUT
+        if (site.id %in% site.pft.links$site) {
```

Comment:
Changed this `if` condition to be checked only when `site.id` is not `NULL`. 

Made the change because I got the below error while testing this function with `site.id` equal to `NULL`.

> **Argument is of length zero in if statement**